src/parse.c

```diff
@@ -4865,7 +4869,29 @@ static tree_t p_interface_function_specification(void)
    }
 
    consume(tRETURN);
-   type_set_result(type, p_type_mark());
+   if (standard() < STD_19) {
+      if (peek_nth(2) == tOF)
```

Comment:
I'd switch this around and always parse the return identifier if `peek_nth(2) == tOF`. That way users who set the wrong standard version will get a single "requires VHDL-2019" error rather than a lot of unexpected token errors.

Reply:
This should be covered with 5c9b104.